### DISCUSSION TITLE
Zigbee Home Automation socket problem

### DISCUSSION THREAD

**Discussion by SmartKobol** [arendst/Tasmota#13233] [Q&A] [answered]
Dear Folks, After a full day search I gave up to find the problem why I cannot add Sonoff ZZbridge to the ZHA. 1. ZBB tasmotised and ZB firmware updated 2. rule1 has been added 3. fix IP has been added to the device, 4. device and linked ZB devices on webui are shown BUT "socket://192.168.0.24:8888 connection failed". How can be the problem solved?

**Comment by sfromis** [comment, marked as answer]
Template updated to ZHA template? Rule activated? Check console output when restarting, preferably with `Weblog 3` for details.

**Reply by SmartKobol** [reply]
Thank you! I will redo both procedures. Shall the connected devices be detached during the updating?

**Reply by sfromis** [reply]
Not sure what you mean by that. Do check the Tasmota log at restart time to debug issues.

**Reply by SmartKobol** [reply]
So _{"NAME":"ZHA ZBBridge","GPIO":[56,208,0,209,59,58,0,0,0,0,0,0,17],"FLAG":0,"BASE":18}_ and _Rule1 ON System#Boot do TCPStart 8888 endon_ added and _restart 1_ typed result see pic and _WebLog 3_ typed result on pic. In HA ZHA still denies 
![7](https://user-images.githubusercontent.com/81106324/135309334-b166eab0-5787-41dd-81f2-21926b82b4b5.JPG)
![9](https://user-images.githubusercontent.com/81106324/135309340-91fdef38-0d8d-42ad-84b1-39ca7c0dc766.JPG)
connection and port speed does not fix at 115200.  N.b. data flow control software or hardware? That happens...

**Reply by sfromis** [reply]
Not running in ZHA mode. Looks like you did not update the template, which includes activating it - checkmark or selecting module 0. And the command you copy/pasted does not include rule activation, either....

I meant for you to *look* at the log, so *you* can see what happens - or not. Your selection of screenshots did just not show the relevant parts....

**Reply by SmartKobol** [reply]
Ok see below:
![11](https://user-images.githubusercontent.com/81106324/135312737-d258a122-77ef-4b4c-99e3-07562a0b0266.JPG)
![12](https://user-images.githubusercontent.com/81106324/135312743-bbd9381c-7559-4cf8-90ae-79f2939504c9.JPG)

**Reply by sfromis** [reply]
Did you read my post? I was *not* asking you for unusable screenshots. To have any chance, you need to activate the template. And it looks like you did not even set the template, like *not at all*. 
https://templates.blakadder.com/howto.html

**Reply by SmartKobol** [reply]
Well, the template was added and activated and the rule has been copied from here: https://zigbee.blakadder.com/Sonoff_ZBBridge.html. Finally the solution rule that has worked for me: _backlog rule1 ON System#Boot do TCPStart 8888 endon; rule 1 1; template {"NAME":"ZHA ZBBridge","GPIO":[56,208,0,209,59,58,0,0,0,0,0,0,17],"FLAG":0,"BASE":18}; module0_ Thank you for you time and sorry for the miscommunication!

**Reply by sfromis** [reply]
What you posted clearly showed that the template was not activated, and most probably not even added at all. Great that you found how how to do it :-)

**Comment by warning939** [comment]
Hi all,
I’ve followed the wiki here: [Sonoff Zigbee Bridge (ZBBridge) Zigbee compatibility](https://zigbee.blakadder.com/Sonoff_ZBBridge.html), but no luck:

10:37:00.142 CMD: Grp 0, Cmnd ‘TCPSTART’, Idx 1, Len 4, Data ‘8888’
10:37:00.146 RSL: RESULT = {“Command”:“Unknown”}

Steps I took:


1. Tasmotizer: tasmota-zbbridge.bin → ok
2. Wifi settings → ok
3. Firmware upgrade " [ncp-uart-nsw_6.7.9_115200.ota](https://github.com/arendst/Tasmota/blob/58289a59f11d5bae419c30a5f1b86fa4a88b806b/tools/fw_SonoffZigbeeBridge_ezsp/ncp-uart-nsw_6.7.9_115200.ota) "–> ok?!
4. Template: {“NAME”:“ZHA ZBBridge”,“GPIO”:[56,208,0,209,59,58,0,0,0,0,0,0,17],“FLAG”:0,“BASE”:18}
5. Rule1 ON System#Boot do TCPStart 8888 endon

But now I have luck with TCPSTART on 8888… :(

Could someone please guide me what to do?

Thank you so much!

**Reply by sfromis** [reply]
Do *check* if you have the right binary flashed, "Program Version" on the "Information" page.

**Reply by Zootallure** [reply]
### OTA Upgrade warning for ZBBridge? 

_**The OTA Url has the default tasmota.bin.gz on startup.**_

I had a working Sonoff ZHA ZBBridge setup using the 'blakadder' instructions, got the wireless zigbee sensors connected via HA, all fine for last few weeks. 
Then I think i did an OTA firmware upgrade (to 12.2.0) , hit the Upgrade button out of habit as i do for the other tasmota devices, and though not 100% sure, i think this messed things up. The TCPStart failed with the {"Command":"Unknown"}  error.

Tried several times to ugrade the uart fw, but kept getting the Invalid File Signature error, tried this on win, mac,linux boxes via the webinterface of the tasmoto bridge, and for different versions (6.7.8, 6.7.9, 6.7.6archived)  no luck, The Template was not the same as the example, some gpio nrs were different, tried resetting this to example, but it got changed back after reboot. 

Then tried to do an local file upgrade of the Tasmo Bridge fw, but that threw a filespace error. 
OTA upgrade of ZBB release showed no errors but did not solve the TCPStart error after going through setup again. 

In the end i reflashed with the downloaded bridge bin using Tasmotizer. After reboot and it then ran the zigbee withouth having to upgrade the uart fw. I guess this is stored in separate uart controller flash mem?  Reconfiguring all for HA, sensors reconnected without fail,  and happy smiles all around. 

So now that i have the correct fw running again, i can see that the OTA URL points to the standard fw, not the bridge, which i think caused the issue of not being able to start TCP.  The only thing i did not try was OTA upgrade of ZBBridge release after having set the module to type 75. Perhaps that would have selected the correct binary in OTA?

![afbeelding](https://user-images.githubusercontent.com/13659322/202451513-ac01c948-0b2b-4cc5-9643-790a048c51d1.png)


Bin comparison zbbridge / default tasmota:

![afbeelding](https://user-images.githubusercontent.com/13659322/202455098-facfa2a7-bce3-4a05-976d-5b6689fe0955.png)

**Reply by sfromis** [reply]
> The only thing i did not try was OTA upgrade of ZBBridge release after having set the module to type 75. Perhaps that would have selected the correct binary in OTA?

No, you could just have set the correct OTA URL on the firmware upgrade page. Would also have worked after having flashed the wrong binary at first.

**Reply by SergioDeMaster** [reply]
If you use OTA update from tasmota.com, you should change url to http://ota.tasmota.com/tasmota/tasmota-zbbridge.bin.gz
It has different firmware for hub and another devices.
